### PR TITLE
Partial festurespec upload support 

### PR DESCRIPTION
**Pull request by riteshkarvaloc**
Users can be allowed to upload feature specs for some columns, the rest can be obtained programmatically.

### CODE REVIEW

File: dkube/sdk/rsrcs/featureset.py [hunk 1]
```diff
@@ -149,20 +149,24 @@ def validate_features(self, dataframe=None, featurespec=None) -> bool:
...
-            metadata["description"] = None
-            metadata["schema"] = str(schema[i])
-            featureset_metadata.append(metadata)
-
-        return featureset_metadata
+            if str(keys[i]) not in added_names:
+                metadata = {}
+                metadata["name"] = str(keys[i])
+                metadata["description"] = None
+                metadata["schema"] = str(schema[i])
```
Comment (songole) [inline dkube/sdk/rsrcs/featureset.py:167]: Is there a way to just upload a new featurespec without any merges? Does the existing upload api work like this?

Comment (riteshkarvaloc) [reply dkube/sdk/rsrcs/featureset.py:167]: If we are not merging it, then the current feature spec will get overwritten and the partial feature spec(where the data scientist wants to describe few columns) will vanish. 

Currently, we don't merge, we overwrite.

**Comment by riteshkarvaloc** [comment]
Added fix for https://github.com/oneconvergence/gpuaas/issues/6544

**Comment by riteshkarvaloc** [comment]
Added the merge flag https://github.com/oneconvergence/dkube/blob/6452-fixes/dkube/sdk/api.py#L805

**Review by songole — APPROVED** [review]

**Pull request merged**